lib/Sema/CSDiagnostics.cpp

```diff
@@ -101,6 +101,52 @@ Expr *FailureDiagnostic::getArgumentExprFor(Expr *anchor) const {
   return nullptr;
 }
 
+ConstraintLocator *
+FailureDiagnostic::getConstraintLocatorFor(Expr *expr) const {
```

Comment:
Thank you! This could be improved further by making it return `Optional<SelectedOverload>` because there is a little use for locator without calling `getOverloadChoiceIfAvailable` right after, so this method would become a `FailureDiagnostic::getOverloadChoiceIfAvailable(Expr *expr)` :)